### PR TITLE
Fix endpoint gateway targets datasource output

### PR DESCRIPTION
**Pull request by ujjwal-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
--- PASS: TestAccIBMISEndpointGatewayTargetsDataSource_basics (31.51s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm 34.487s
```

### CODE REVIEW

File: ibm/data_source_ibm_is_endpoint_gateway_targets.go [hunk 1]
```diff
@@ -157,9 +157,38 @@ func dataSourceIBMISEndpointGatewayTargetsRead(context context.Context, d *schem
...
+	searchResult, err := globalSearchClient.Searches().PostQuery(getSearchOptions)
+	if err != nil {
+		log.Printf("[DEBUG] PostQuery on globalSearchApi failed %s", err)
+		return diag.FromErr(err)
+	}
+	searchItems := searchResult.Items
+	for _, item := range searchItems {
+		info := map[string]interface{}{}
+		info[isVPEResourceName] = item.Name
+		info[isVPEResourceType] = item.Name
```
Comment (astha-jain) [inline ibm/data_source_ibm_is_endpoint_gateway_targets.go:184]: isVPEResourceName and isVPEResourceType both are getting item.Name? 

type should be "provider_cloud_service"

File: ibm/data_source_ibm_is_endpoint_gateway_targets.go [hunk 2]
```diff
@@ -157,9 +157,38 @@ func dataSourceIBMISEndpointGatewayTargetsRead(context context.Context, d *schem
...
+		Query:  queryString,
+		Fields: fields,
+	}
+	globalSearchClient, err := meta.(ClientSession).GlobalSearchAPI()
+	if err != nil {
+		return diag.FromErr(err)
 	}
+	searchResult, err := globalSearchClient.Searches().PostQuery(getSearchOptions)
+	if err != nil {
+		log.Printf("[DEBUG] PostQuery on globalSearchApi failed %s", err)
```
Comment (astha-jain) [inline ibm/data_source_ibm_is_endpoint_gateway_targets.go:177]: Lets specify the operation where this call is being made - Customer won't check into the code to understand the error `PostQuery on globalSearchApi retrieving endpoint gateway targets failed`

**Review by astha-jain — APPROVED** [review]

File: ibm/data_source_ibm_is_endpoint_gateway_targets.go (hunk 1 above)
Comment (ujjwal-ibm) [reply ibm/data_source_ibm_is_endpoint_gateway_targets.go]: this is for search items, have updated it now after bluemix go fix

File: ibm/data_source_ibm_is_endpoint_gateway_targets.go (hunk 2 above)
Comment (ujjwal-ibm) [reply ibm/data_source_ibm_is_endpoint_gateway_targets.go]: Added query string in the log

**Comment by ujjwal-ibm** [comment]
moving to upstream

**Pull request closed without merging**